### PR TITLE
OSPF improvements

### PR DESCRIPTION
**Pull request by dsgagi**
- Fixed ospf port listing, as current query only returned ospf ports for last AreaID in `area` foreach loop.
- Added `Area ID` column to Ports table.
- Removed whitespaces.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Doesn't this make the outer foreach useless?

You grab all the ospf entries every loop...

**Comment by dsgagi** [comment]
Do you mean `area` or `instance` foreach loop?

I agree, the code is not so easy to follow :)

Each ospf instance has it's own hidden row (with areas, ports and neighbor columns). I guess it's coded this way for rare cases when there are more ospf instances/processes running on the device. 

As for `area` loop, it doesn't include `ospfport` loop. It includes only the code below.

```
foreach (dbFetchRows('SELECT * FROM `ospf_areas` WHERE `device_id` = ?', array($device['device_id'])) as $area) {
        $area_port_count         = dbFetchCell('SELECT COUNT(*) FROM `ospf_ports` WHERE `device_id` = ? AND `ospfIfAreaId` = ?', array($device['device_id'], $area['ospfAreaId']));
        $area_port_count_enabled = dbFetchCell("SELECT COUNT(*) FROM `ospf_ports` WHERE `ospfIfAdminStat` = 'enabled' AND `device_id` = ? AND `ospfIfAreaId` = ?", array($device['device_id'], $area['ospfAreaId']));

        echo '
                      <tbody>
                        <tr>
                          <td>' . $area['ospfAreaId'] . '</td>
                          <td>' . $area_port_count . '(' . $area_port_count_enabled . ')</td>
                          <td><span class="label label-' . $status_color . '">' . $instance['ospfAdminStat'] . '</span></td>
                        </tr>
                      </tbody>';
}
```

File: html/pages/device/routing/ospf.inc.php [hunk 1]
```diff
@@ -94,24 +94,26 @@
                           <th>Port Type</th>
                           <th>Port State</th>
                           <th>Status</th>
+                          <th>Area ID</th>
                         </tr>
                       </thead>
                   </div>';
-    foreach (dbFetchRows("SELECT * FROM `ospf_ports` AS O, `ports` AS P WHERE O.`ospfIfAdminStat` = 'enabled' AND O.`device_id` = ? AND O.`ospfIfAreaId` = ? AND P.port_id = O.port_id", array($device['device_id'], $area['ospfAreaId'])) as $ospfport) {
```
[the remark this answers is not in the document]
Comment (dsgagi) [reply html/pages/device/routing/ospf.inc.php]: Actually, this foreach loop is not included inside the foreach loop for the $area variable.

The end result is that port info is generated only for one ospf area (last one from the loop before).

**Comment by murrant** [comment]
Web files have been moved to includes/html, you will need to update your PR, sorry for the trouble.  Let us know if you need any help.

**Comment by dsgagi** [comment]
Hello. Just returned from the vacation and reapplied the patch after git rebase.

First time doing the rebase. I'm hoping I did it right. :)

**Comment by PipoCanaja** [comment]
Hi @dsgagi 
I am not so sure the rebase was done the usual way :) You end up with 3000+ file changed from master, and we cannot see what was really changed with your PR. 

Could you try to create a new clean branch on your dev-server with only your changed files, and create a new Pull Request from it ?

Usually, you don't even need to rebase a dev branch. But anyway, if you need to do a rebase, try the following steps:
```
## Switch to master
git checkout master

## Update Master and push it to your github (to keep it in sync with librenms master)
git pull upstream master
git push origin master

## Switch to your branch
git checkout <yourBranch>

## Rebase it
git rebase master

## Push your branch to your github
git push origin <yourBranch>
```

**Comment by dsgagi** [comment]
Hi @PipoCanaja,

I tried those same steps actually (found the steps in librenms documentation), but unfortunately ended with this situation in the PR :) . I guess that moving `html` folder under `includes` folder was too much, so my PR caught all the other changes too. Very possible that I just F-ed up during rebase :)

Will create a new branch/PR and reference it here after that.

**Comment by dsgagi** [comment]
Opened new PR - #10206

**Pull request closed without merging**